transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionNotifierTestCase.java

```diff
@@ -0,0 +1,218 @@
+            protected boolean test() throws Exception
+            {
+                verify(mockNotifier, times(1)).onSuccess(any(RetryContext.class));
+                return true;
+            }
+
+            @Override
+            public String describeFailure()
+            {
+                return "Success notification wasn't triggered";
+            }
+        });
+    }
+
+    private void verifyOnFailureNotification()
+    {
+        PollingProber pollingProber = new PollingProber(POLLING_FREQUENCY * 5, POLLING_FREQUENCY);
+        pollingProber.check(new JUnitProbe()
+        {
+            @Override
+            protected boolean test() throws Exception
+            {
+                verify(mockNotifier, atLeast(1)).onFailure(any(RetryContext.class), any(Throwable.class));
+                return true;
+            }
+
+            @Override
+            public String describeFailure()
+            {
+                return "Failure notification wasn't triggered";
+            }
+        });
+    }
+
+    private void testServerConnected() throws Exception
+    {
+        runFlow("main-test");
+        verifyOnSuccessNotification();
+    }
+
+    private void testServerDisconnected()
+    {
+        try
+        {
+            runFlow("main-test");
+        }
+        catch (Exception e)
+        {
+            verifyOnFailureNotification();
+        }
+    }
+
+
+    public static class MuleFtpServer
+    {
+        private FtpServer server;
+        FtpServerFactory serverFactory;
+        ListenerFactory factory;
+        public MuleFtpServer(int port, long delay)
+        {
+
+            serverFactory = new FtpServerFactory();
+            factory = new ListenerFactory();
+            factory.setPort(port);
+            serverFactory.addListener("default", factory.createListener());
+            serverFactory.addListener("default", factory.createListener());
+            PropertiesUserManagerFactory userManagerFactory = new PropertiesUserManagerFactory();
+            serverFactory.setUserManager(userManagerFactory.createUserManager());
+            BaseUser user = new BaseUser();
+            user.setName("mule-test");
+            user.setPassword("mule-test");
+            String currentDir = System.getProperty("user.dir");
+            user.setHomeDirectory(currentDir);
+            List<Authority> authorities = new ArrayList<Authority>();
+            authorities.add(new WritePermission());
+            user.setAuthorities(authorities);
+            try
+            {
+                serverFactory.getUserManager().save(user);
+            }
+            catch (FtpException e)
+            {
+                e.printStackTrace();
+            }
+            server = serverFactory.createServer();
+        }
+
+        public void start()
+        {
+            try
+            {
+                server.start();
+            }
+            catch (FtpException e)
+            {
+                e.printStackTrace();
```

Comment:
Do not print the exception, wrap the exception on a RuntimeException and throw it